pkg/migrate/migrate.go

```diff
@@ -46,6 +49,10 @@ func migrate(ctx context.Context, log logr.Logger, c client.Client) error {
 		// not critical for the operator operation, safer to ignore
 		log.V(consts.LogLevelWarning).Info("ignore error during whereabouts CronJob removal")
 	}
+	if err := removeStateLabelFromNVIpamConfigMap(ctx, log, c); err != nil {
```

Comment:
Why do we need to return error if we ignore it? We've got logging inside  removeStateLabelFromNVIpamConfigMap so no need to return an error

Reply:
I followed the same pattern as `removeWhereaboutsIPReconcileCronJob`.
WDYT?

Reply:
I think we should keep it as is. `removeStateLabelFromNVIpamConfigMap` function do the action and an error can happen during this action - it is fair to return the error from the function.  It is up to caller to decide what to do with the error.

Reply:
and this should not block the PR for sure

Reply:
OK, I have one more thing. It is pretty safe to run `removeWhereaboutsIPReconcileCronJob` in a "best effort" mode and ignore any error if it happens (we will have an unneeded Job in the cluster, but this will not break anything). But with `removeStateLabelFromNVIpamConfigMap`, we are in a bit different situation - if we fail to remove the label, we will probably lose all nv-ipam allocations. Maybe we should crash in this case and rely on k8s to restart the Pod? WDYTH?

Reply:
Agreed, returning error to main.